pkg/apis/core/v1/package_webhook.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"fmt"
+
+	"github.com/dustin/go-humanize"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	ferror "github.com/fission/fission/pkg/error"
+	"github.com/fission/fission/pkg/utils/loggerfactory"
+)
+
+// log is for logging in this package.
+var packagelog = loggerfactory.GetLogger().Named("package-resource")
+
+func (r *Package) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/mutate-fission-io-v1-package,mutating=true,failurePolicy=fail,sideEffects=None,groups=fission.io,resources=packages,verbs=create;update,versions=v1,name=mpackage.fission.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &Package{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *Package) Default() {
+	packagelog.Debug("default", zap.String("name", r.Name))
+	r.Status.BuildStatus = BuildStatusSucceeded
```

Comment:
We should check if the build status is not set and set to build status pending.

We should check if the build status falls in one of the valid states in the validation hook.